server/log_rpc_server_test.go

```diff
@@ -526,6 +526,118 @@ func TestGetLeavesByHash(t *testing.T) {
 	}
 }
 
+func TestGetLeavesByLeafValueHashInvalidHash(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockStorage := storage.NewMockLogStorage(ctrl)
+
+	server := NewTrillianLogRPCServer(mockStorageProviderfunc(mockStorage), fakeTimeSource)
+
+	// This request includes an empty hash, which isn't allowed
+	resp, err := server.GetLeavesByLeafValueHash(context.Background(), &getByHashRequestBadHash)
+
+	// Should have succeeded at RPC level
+	if err != nil {
+		t.Fatalf("GetLeavesByLeafValueHash() = %v", err)
+	}
+
+	// And failed at app level
+	if got, want := resp.Status.StatusCode, trillian.TrillianApiStatusCode_ERROR; got != want {
+		t.Fatalf("Bad status got: %v, want: %v", got, want)
+	}
+}
+
+func TestGetLeavesByLeafValueHashBeginFails(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockStorage := storage.NewMockLogStorage(ctrl)
+	mockTx := storage.NewMockLogTX(ctrl)
+
+	mockStorage.EXPECT().Begin().Return(mockTx, errors.New("TX"))
+
+	server := NewTrillianLogRPCServer(mockStorageProviderfunc(mockStorage), fakeTimeSource)
+
+	if _, err := server.GetLeavesByLeafValueHash(context.Background(), &getByHashRequest1); err == nil || !strings.Contains(err.Error(), "TX") {
+		t.Fatalf("Wrong error response when begin failed: %v", err)
+	}
+}
+
+func TestGetLeavesByLeafValueHashStorageFails(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	test := newParameterizedTest(ctrl, "GetLeavesByLeafValueHash",
+		func(t *storage.MockLogTX) {
+			t.EXPECT().GetLeavesByLeafValueHash([][]byte{[]byte("test"), []byte("data")}, false).Return([]trillian.LogLeaf{}, errors.New("STORAGE"))
+		},
+		func(s *TrillianLogRPCServer) error {
+			_, err := s.GetLeavesByLeafValueHash(context.Background(), &getByHashRequest1)
+			return err
+		})
+
```

Comment:
(pre-existing) I have problems understanding what's going on with these tests -- there's quite a lot of action-at-a-distance and mixtures of constants and explicit values.  Maybe discuss offline.